src/client/tsconfig.json

```
@@ -1,10 +1,12 @@
 {
-  "extends": "../tsconfig.json",
+  "extends": "../../tsconfig.json",
   "compilerOptions": {
     "baseUrl": ".",
     "outDir": "../../dist/client",
     "target": "esnext",
     "module": "esnext",
+    "declaration": true,
+    "declarationDir": "../../dist/client-types",
```

This should be `client` instead of `client-types`?

```suggestion
    "declarationDir": "../../dist/client",
```

---

Wait not really, we need `client-types` in rollup config... However, this raises

```
[!] (plugin cleanup) Error: ENOENT: no such file or directory, stat '/Users/kia/Code/vitepress-next/dist/client-types'
```

When running `pnpm run docs`.

---

OK this is happening because on `dev` command, client build executes, and then node build (rollup) executes in parallel and `client-types` folder gets deleted becaue of `cleanup` hook in rollup config. So we should first check for the folder existense and then perform delete on the folder I guess.

So maybe we just do this?

```ts
// rollup.config.ts
plugins: [
  dts(),
  {
    name: 'cleanup',
    async closeBundle() {
      try {
        await fs.rm(r('dist/client-types'), { recursive: true })
      } catch {}
    }
  }
]
```